The ``BaseGeometry`` module
=================================

using ``BaseGeometry`` class
------------------------

	>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry

	>>> m = __import__("7-base_geometry").__doc__
	>>> len(m) > 1
	True

	>>> c = __import__("7-base_geometry").BaseGeometry.__doc__
	>>> len(c) > 1
	True

	>>> mod1 = __import__("7-base_geometry").BaseGeometry.area.__doc__
	>>> len(mod1) > 1
	True
	>>> mod2 = __import__("7-base_geometry").BaseGeometry.integer_validator.__doc__
	>>> len(mod2) > 1
	True
	>>> bg = BaseGeometry()
	>>> type(bg)
	<class '7-base_geometry.BaseGeometry'>

	>>> bg.integer_validator("num", 16)

	>>> bg.integer_validator("num", -1)
	Traceback (most recent call last):
	...
	ValueError: num must be greater than 0

	>>> bg.integer_validator("my_num", 0)
	Traceback (most recent call last):
	...
	ValueError: my_num must be greater than 0

	>>> bg.area()
	Traceback (most recent call last):
	...
	Exception: area() is not implemented

	>>> bg.integer_validator("float", 3.04)
	Traceback (most recent call last):
	...
	TypeError: float must be an integer

	>>> bg.integer_validator("me", (2, 3))
	Traceback (most recent call last):
	...
	TypeError: me must be an integer

	>>> bg.integer_validator("me", "4")
	Traceback (most recent call last):
	...
	TypeError: me must be an integer

	>>> bg.integer_validator("you", [4])
	Traceback (most recent call last):
	...
	TypeError: you must be an integer

	>>> bg.integer_validator("fr", {4: 21})
	Traceback (most recent call last):
	...
	TypeError: fr must be an integer

	>>> bg.integer_validator("person", True)
	Traceback (most recent call last):
	...
	TypeError: person must be an integer

	>>> bg.integer_validator("what", None)
	Traceback (most recent call last):
	...
	TypeError: what must be an integer

	>>> bg.integer_validator()
	Traceback (most recent call last):
	...
	TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'

	>>> bg.integer_validator("age")
	Traceback (most recent call last):
	...
	TypeError: integer_validator() missing 1 required positional argument: 'value'
